src/plugins/discover/public/application/main/services/discover_data_state_container.ts

```diff
@@ -219,6 +224,27 @@ export function getDataStateContainer({
         getInternalState,
         savedSearch: getSavedSearch(),
         useNewFieldsApi: !uiSettings.get(SEARCH_FIELDS_FROM_SOURCE),
+      };
+
+      if (options.fetchMore) {
+        abortControllerFetchMore?.abort();
```

Comment:
Wouldn't we want to cancel both kinds of requests regardless of the case?

Reply:
@weltenwort Right! Updated via https://github.com/elastic/kibana/pull/157241/commits/5d10203af084e0b4542704bef6124c4d6acb6796